examples/hidparse.rs

```diff
@@ -0,0 +1,157 @@
+//! HID Report Descriptor Parser Utility
+//!
+//! Simple command line utility that supports parsing descriptors and printing the results.
+//!
+//! Demonstrates the usage of the [`hidparser`] crate.
+//!
+//! # Usage
+//!
+//! `cargo run --example hidparse -- --path .\examples\samples\boot_keyboard.bin`
+//!
+//! ## License
+//!
+//! Copyright (c) Microsoft Corporation. All rights reserved.
+//!
+//! SPDX-License-Identifier: BSD-2-Clause-Patent
+//!
+
+use std::{
+    error::Error,
+    fs::{self, File},
+    io::BufReader,
+    iter,
+    path::PathBuf,
+  };
+  
+  use clap::{Parser, ValueEnum};
+  use hidparser::{parse_report_descriptor, ReportDescriptor, ReportField};
+  use serde::{Deserialize, Serialize};
+  use serde_json::{Map, Value};
+  
+  #[derive(Parser, Debug, Clone, PartialEq, Eq, ValueEnum)]
+  enum ReportType {
+    Input,
+    Output,
+    Feature,
+  }
+  
+  /// Simple command line utility that supports parsing descriptors and printing the results.
+  ///
+  /// Copyright (c) Microsoft Corporation. All rights reserved.
+  ///
+  /// SPDX-License-Identifier: BSD-2-Clause-Patent
+  #[derive(Parser, Debug)]
+  struct Arguments {
+    /// The path containing descriptor binary file.
+    #[arg(short, long)]
+    path: std::path::PathBuf,
+  
+    /// Type of report to list
+    #[arg(short, long)]
+    report_type: Option<ReportType>,
+  
+    /// Id of report to list
+    #[arg(short = 'i', long)]
+    report_id: Option<u32>,
+  }
+  
+  #[derive(Serialize, Deserialize, Debug)]
+  #[serde(rename_all = "PascalCase")]
+  struct UsageId {
+    id: u16,
+    name: String,
+    kinds: Vec<String>,
+  }
+  
+  #[derive(Serialize, Deserialize, Debug)]
+  #[serde(rename_all = "PascalCase")]
+  struct UsagePage {
+    kind: String,
+    id: u16,
+    name: String,
+    usage_ids: Vec<UsageId>,
+    #[serde(skip)]
+    _usage_id_generator: Map<String, Value>,
+  }
+  
+  #[derive(Serialize, Deserialize, Debug)]
+  #[allow(non_snake_case)]
+  struct HidUsageTable {
+    UsageTableVersion: u8,
```

Comment:
Any particular reason we are not using snake case here? I'm not bothered about it one way or the other, just curious